app/models/agents/http_status_agent.rb

```diff
@@ -82,7 +82,8 @@ def check_this_url(url, local_headers)
 
       # Deal with failures
       if measured_result.result
-        payload.merge!({ 'response_received' => true, 'status' => current_status })
+        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.to_hash[:url]
```

Comment:
`measured_result.result.url` doesn't work anywhere. `measured_result.url` works in RSpec, weirdly, but not in actual Rails. ????

In any case, I ended up just mocking `to_hash` (it's basically an alias to `to_h`). So specs pass now! \o/